cmd/kwild/server/build.go

```diff
@@ -450,31 +540,92 @@ func initAccountRepository(d *coreDependencies, tx sql.Tx) {
 	}
 }
 
-func buildSnapshotter() *snapshots.SnapshotStore {
-	return nil
-	// TODO: Uncomment when we have statesync ready
-	// cfg := d.cfg.AppCfg
-	// if !cfg.SnapshotConfig.Enabled {
-	// 	return nil
-	// }
+func buildSnapshotter(d *coreDependencies) *statesync.SnapshotStore {
+	cfg := d.cfg.AppCfg
+	if !cfg.Snapshots.Enabled {
+		return nil
+	}
 
-	// return snapshots.NewSnapshotStore(cfg.SqliteFilePath,
-	// 	cfg.SnapshotConfig.SnapshotDir,
-	// 	cfg.SnapshotConfig.RecurringHeight,
-	// 	cfg.SnapshotConfig.MaxSnapshots,
-	// 	snapshots.WithLogger(*d.log.Named("snapshotStore")),
-	// )
+	dbCfg := &statesync.DBConfig{
+		DBUser: cfg.DBUser,
+		DBPass: cfg.DBPass,
+		DBHost: cfg.DBHost,
+		DBPort: cfg.DBPort,
+		DBName: cfg.DBName,
+	}
+
+	snapshotCfg := &statesync.SnapshotConfig{
+		SnapshotDir:     cfg.Snapshots.SnapshotDir,
+		RecurringHeight: cfg.Snapshots.RecurringHeight,
+		MaxSnapshots:    int(cfg.Snapshots.MaxSnapshots),
+	}
+
+	ss, err := statesync.NewSnapshotStore(snapshotCfg, dbCfg, *d.log.Named("snapshot-store"))
+	if err != nil {
+		failBuild(err, "failed to build snapshot store")
+	}
+	return ss
 }
 
-func buildBootstrapper() *snapshots.Bootstrapper {
-	return nil
-	// TODO: Uncomment when we have statesync ready
-	// rcvdSnapsDir := filepath.Join(d.cfg.RootDir, rcvdSnapsDirName)
-	// bootstrapper, err := snapshots.NewBootstrapper(d.cfg.AppCfg.SqliteFilePath, rcvdSnapsDir)
-	// if err != nil {
-	// 	failBuild(err, "Bootstrap module initialization failure")
-	// }
-	// return bootstrapper
+func buildStatesyncer(d *coreDependencies) *statesync.StateSyncer {
+	if !d.cfg.ChainCfg.StateSync.Enable {
+		return nil
+	}
+
+	cfg := d.cfg.AppCfg
+
+	dbCfg := &statesync.DBConfig{
+		DBUser: cfg.DBUser,
+		DBPass: cfg.DBPass,
+		DBHost: cfg.DBHost,
+		DBPort: cfg.DBPort,
+		DBName: cfg.DBName,
+	}
+
+	providers := strings.Split(d.cfg.ChainCfg.StateSync.RPCServers, ",")
+
+	if len(providers) == 0 {
+		failBuild(nil, "failed to configure state syncer, no remote servers provided.")
+	}
+
+	if len(providers) == 1 {
+		d.log.Warn("multiple state sync providers are not supported yet, using the first one")
+		// Duplicating the same provider to satisfy cometbft statesync requirement of having at least 2 providers.
+		d.cfg.ChainCfg.StateSync.RPCServers += "," + providers[0]
+	}
+
+	configDone := false
+	for _, p := range providers {
+		clt, err := statesync.ChainRPCClient(p)
+		if err != nil {
+			continue
+		}
+
+		// Try to fetch the status of the remote server.
+		res, err := clt.Commit(d.ctx, nil)
```

Comment:
`clt.Header` would be sufficient.  We don't need the commit evidence, only the header (`Hash()` method give the hash)